src/init.cpp

```diff
@@ -963,7 +964,8 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     mempool.SetMempoolCostLimit(mempoolTotalCostLimit, mempoolEvictionMemorySeconds);
 
     fCheckBlockIndex = GetBoolArg("-checkblockindex", chainparams.DefaultConsistencyChecks());
-    fCheckpointsEnabled = GetBoolArg("-checkpoints", DEFAULT_CHECKPOINTS_ENABLED);
+    fIBDSkipTxVerification = GetBoolArg("-ibdskiptxverification", DEFAULT_IBD_SKIP_TX_VERIFICATION);
+    fCheckpointsEnabled = GetBoolArg("-checkpoints", DEFAULT_CHECKPOINTS_ENABLED) || fIBDSkipTxVerification;
```

Comment:
```suggestion
    fCheckpointsEnabled = GetBoolArg("-checkpoints", DEFAULT_CHECKPOINTS_ENABLED);
```

This is not the correct way to handle this, because it silently ignores conflicting user configuration (the default for `-checkpoints` is true, thus we could only reach the second half of this condition if the user is making an explicit configuration choice to disable checkpoints). This could lead to the user running a node that they think is verifying all transactions, but in fact is not. As I pointed to in my original comment, the necessary change should go just after here:

https://github.com/zcash/zcash/blob/e2d3533de337cf74a2e84b36712cdb6286720e62/src/init.cpp#L914-L923

It should similarly check for the conflicting configuration of `-ibdskiptxverification` and `-nocheckpoints`, and return an `InitError`.

Reply:
Ah, okay, I didn't understand that detail.